sdk-tests/src/test/java/io/dapr/it/spring/data/MySQLDaprKeyValueTemplateIT.java

```diff
@@ -0,0 +1,246 @@
+import java.util.Optional;
+
+import static io.dapr.it.spring.data.DaprSpringDataConstants.STATE_STORE_NAME;
+import static io.dapr.it.spring.data.DaprSpringDataConstants.BINDING_NAME;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Integration tests for {@link MySQLDaprKeyValueTemplateIT}.
+ */
+@SuppressWarnings("AbbreviationAsWordInName")
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = TestDaprSpringDataConfiguration.class)
+@Testcontainers
+@Tag("testcontainers")
+public class MySQLDaprKeyValueTemplateIT {
+  private static final String STATE_STORE_DSN = "mysql:password@tcp(mysql:3306)/";
+  private static final String BINDING_DSN = "mysql:password@tcp(mysql:3306)/dapr_db";
+  private static final Map<String, String> STATE_STORE_PROPERTIES = createStateStoreProperties();
+
+  private static final Map<String, String> BINDING_PROPERTIES = Collections.singletonMap("url", BINDING_DSN);
+
+  private static final Network DAPR_NETWORK = Network.newNetwork();
+
+  @Container
+  private static final MySQLContainer<?> MY_SQL_CONTAINER = new MySQLContainer<>("mysql:5.7.34")
+      .withNetworkAliases("mysql")
+      .withDatabaseName("dapr_db")
+      .withUsername("mysql")
+      .withPassword("password")
+      .withExposedPorts(3306)
+      .withNetwork(DAPR_NETWORK);
+
+  private static final WaitStrategy DAPR_CONTAINER_WAIT_STRATEGY = Wait.forHttp("/v1.0/healthz")
+      .forPort(3500)
+      .forStatusCodeMatching(statusCode -> statusCode >= 200 && statusCode <= 399);
+
+  @Container
+  private static final DaprContainer DAPR_CONTAINER = new DaprContainer("daprio/daprd:1.13.2")
+      .withAppName("mysql-dapr-app")
+      .withNetwork(DAPR_NETWORK)
+      .withComponent(new Component(STATE_STORE_NAME, "state.mysql", "v1", STATE_STORE_PROPERTIES))
+      .withComponent(new Component(BINDING_NAME, "bindings.mysql", "v1", BINDING_PROPERTIES))
+      .withDaprLogLevel(DaprLogLevel.DEBUG)
+      .withLogConsumer(outputFrame -> System.out.println(outputFrame.getUtf8String()))
+      .waitingFor(DAPR_CONTAINER_WAIT_STRATEGY)
+      .dependsOn(MY_SQL_CONTAINER);
+
+  @DynamicPropertySource
+  static void daprProperties(DynamicPropertyRegistry registry) {
+    DAPR_CONTAINER.start();
+    registry.add("dapr.grpc.port", DAPR_CONTAINER::getGrpcPort);
+    registry.add("dapr.http.port", DAPR_CONTAINER::getHttpPort);
+  }
+
+  private static Map<String, String> createStateStoreProperties() {
+    Map<String, String> result = new HashMap<>();
+
+    result.put("keyPrefix", "name");
+    result.put("schemaName", "dapr_db");
+    result.put("actorStateStore", "true");
+    result.put("connectionString", STATE_STORE_DSN);
+
+    return result;
+  }
+
+  @Autowired
+  private DaprClient daprClient;
+
+  @Autowired
+  private DaprKeyValueTemplate keyValueTemplate;
+
+  @BeforeEach
+  public void waitSetup() {
+    daprClient.waitForSidecar(1000).block();
```

Comment:
@artursouza we are doing this now, but I am seriously thinking to move this down into the DaprKeyValueTemplate so the user doesn't need to worry about handling the `daprClient` would it make sense to run `daprClient.waitForSidecar(1000).block()` before each Dapr API operation? Is that the intended use for all the operations?